rudder-core/src/main/resources/hooks.d/node-policy-generated/readme.txt

```diff
@@ -0,0 +1,44 @@
+
+Generallity about Rudder Hooks
+==============================
+
+Hooks must be executable file, typically scripts, whatever the language
+as long they are self-executable (i.e: correct #! definition). 
+
+Hooks must be name with the extension ".hook". Any other extension will 
+be ignore (which allow to put other file in these directory, like a 
+readme, for example). 
+
+Hooks are executed sequentially, in alpha-numeric order. We encourage
+you to use the patter "NN-hookname.ext.hook", with NN a number like
+"05", "20", etc. 
+
+About node-policy-generated/ hooks
+==================================
+
+When/What ?
+-----------
+
+This directory contains hooks executed after poplicies are fully 
+generated for node, but before these new policies replace the old
+one (technically, before we move /var/rudder/share/$NODEID/rules.new 
+to /var/rudder/share/$NODEID/rules).
+
+Typically, these hooks proceed to sanity checks on the new policies
+(ex: cf-promises), update permission on files, or interect with
+external services using knowledged from the generated policies 
+(ex: send node-properties JSON file to a third party service). 
+
+Parameters ?
+------------
+
+- #1: generation datetime: ISO-8601 date/time that identify that policy generation start 
+- #2: the nodeId
+- #3: new policies directory (ie /var/rudder/share/$NODEID/rules.new)
+- #4: agent type (cfengine-nova or cfengine-community)
+
+Technically, you could infer #3 from #2, but it is just simpler to
```

Comment:
actually, iy's pretty hard with relays to infer #3 from #2, unless you are doing API calls for each node